### ISSUE TITLE
Implement java-control-plane example

### ISSUE THREAD

**Issue by zachmandeville** [ii/xds-test-harness#56]
This is a second example of an xds server that integrates the test harness adapter. 

We will use the [java-control-plane](https://github.com/envoyproxy/java-control-plane) as our target implementation.

This work involves:
- [x] learning _enough_ java for the project
- [x] setting up a build system and development environment for implementation
- [x] implementing an xds server using the java-control-plane library
- [x] using build system to generate java code from our adapter.proto files
- [x] implement basic working adapter
- [x] integrate example into this repo
- [x] pass first subscribing test with example implementation using SOTW
- [x] pass all subscribing tests, SOTW
- [x] pass first unsubscribing test, SOTW
- [x] pass all unsubscribing tests, SOTW
- [ ] pass suite using incremental, non-aggregated variant
- [ ] pass suite using non-incremental, aggregated variant
- [ ] pass suite using incremental, aggregated variant
- [ ] add gradle tasks to package example implementation for easy executables

- [ ] document its usage

**Comment by zachmandeville** [comment]
Using gradle, set up a build and development environment via intelliJ.

I have an xDS server set up integrated with an adapter.  The adapter implements our api, but in the most rudimentary way possible.  It simply returns "success" no matter what is given to it.  This is of course not usable as a real adapter, but helps us make sure our general connectivity is working.

I than ran the first subscribing test:

```
  Scenario Outline: [<xDS>] The service should send all resources on a wildcard request.
    Given a target setup with service <xDS>, resources <resources>, and starting version <v1>
    When the Client does a wildcard subscription to <xDS>
    Then the Client receives the resources <expected> and version <v1> for <xDS>
    And the resources <expected> and version <v1> for <xDS> came in a single response
    And the service never responds more than necessary
```

The test failed on the step `Then the Client receives the resources <expected> and version <v1> for <xDS>`, which we expect.

However, to get to the test, the runner must be able to connect to an adapter and a target.  That the test ran at all means the connections worked.  Also, this test passed the first two steps.  The first step sends a request to the adapter and checks that it gets a valid response back.  The second step subscribes to the target and makes sure no error is thrown.  This means that not only can our runner connect to the java implementation, but the two servers are working as expected.

Next up is implementing the adapter in earnest and getting the remainder of these steps to pass.

**Comment by zachmandeville** [comment]
Made good progress today.  Implemented all the methods for the adapter with the java control plane and all subscribing and unsubscribing tests pass for the non-incremental, non-aggregated variant.

It is not the most beautiful java, but it works.  

I opened a PR (#57 ) with the code so far. 

The work is far from done.  I want to make sure all tests pass for all variants, or that we can at least confirm that any failing tests are not on our side.  I also want the code to be as simple and clean as possible.  There is a lot of repetition coming from my inexperience with java convention.  I also need to set up better build tasks and logging. At the moment, this implementation is best run from within intelliJ. It would be good to have it packaged as an executable.

Regardless, a solid leap forward today, I  think!
